mongo/datadog_checks/mongo/api.py

```diff
@@ -0,0 +1,104 @@
+from pymongo import MongoClient, ReadPreference
+from pymongo.errors import PyMongoError
+
+from datadog_checks.mongo.common import MongosDeployment, ReplicaSetDeployment, StandaloneDeployment
+
+# The name of the application that created this MongoClient instance. MongoDB 3.4 and newer will print this value in
+# the server log upon establishing each connection. It is also recorded in the slow query log and profile collections.
+DD_APP_NAME = 'datadog-agent'
+
+
+class MongoApi(MongoClient):
```

Comment:
(Nit) Do we want to prefer composition over inheritance here? I feel like it could be clearer and more maintainable to explicitly define the contract between this API class and the check / collectors, rather than merging it with `MongoClient`. Something like this:

```python
class MongoAPI(object):
    def __init__(self, ...):
        ...
        self._client = MongoClient(...)

    ...  # Explicit methods here
```

Reviewing the code it seems we'd just need to explicitly define two extra methods:

- `__getitem__(name: str) -> Any`
- `.server_info() -> dict`